src/common/src/monitor/connection.rs

```diff
@@ -0,0 +1,573 @@
+        let labels = ["connection_type", "uri"];
+        let connection_count = register_int_gauge_vec_with_registry!(
+            "connection_count",
+            "The number of current existing connection",
+            &labels,
+            registry,
+        )
+        .unwrap();
+
+        let connection_create_rate = register_int_counter_vec_with_registry!(
+            "connection_create_rate",
+            "Rate on creating new connection",
+            &labels,
+            registry,
+        )
+        .unwrap();
+
+        let connection_err_rate = register_int_counter_vec_with_registry!(
+            "connection_err_rate",
+            "Error rate on creating new connection",
+            &labels,
+            registry,
+        )
+        .unwrap();
+
+        let read_rate = register_int_counter_vec_with_registry!(
+            "connection_read_rate",
+            "Read rate of a connection",
+            &labels,
+            registry,
+        )
+        .unwrap();
+
+        let reader_count = register_int_gauge_vec_with_registry!(
+            "connection_reader_count",
+            "The number of current existing reader",
+            &labels,
+            registry,
+        )
+        .unwrap();
+
+        let write_rate = register_int_counter_vec_with_registry!(
+            "connection_write_rate",
+            "Write rate of a connection",
+            &labels,
+            registry,
+        )
+        .unwrap();
+
+        let writer_count = register_int_gauge_vec_with_registry!(
+            "connection_writer_count",
+            "The number of current existing writer",
+            &labels,
+            registry,
+        )
+        .unwrap();
+
+        let io_err_rate = register_int_counter_vec_with_registry!(
+            "connection_io_err_rate",
+            "IO err rate of a connection",
+            &["connection_type", "uri", "op_type", "error_kind"],
+            registry,
+        )
+        .unwrap();
+
+        Self {
+            connection_count,
+            connection_create_rate,
+            connection_err_rate,
+            read_rate,
+            reader_count,
+            write_rate,
+            writer_count,
+            io_err_rate,
+        }
+    }
+}
+
+pub struct TcpConfig {
+    pub tcp_nodelay: bool,
+    pub keepalive_duration: Option<Duration>,
+}
+
+pub fn monitored_hyper_https_connector(
+    connection_type: impl Into<String>,
+    config: TcpConfig,
+) -> MonitoredConnection<HttpConnector, MonitorNewConnectionImpl> {
+    let mut http = HttpConnector::new();
+    http.enforce_http(false);
+    http.set_nodelay(config.tcp_nodelay);
+    http.set_keepalive(config.keepalive_duration);
+
+    monitor_connector(http, connection_type)
+}
+
+pub fn monitor_connector<C>(
+    connector: C,
+    connection_type: impl Into<String>,
+) -> MonitoredConnection<C, MonitorNewConnectionImpl> {
+    let connection_type = connection_type.into();
+    info!(
```

Comment:
Is this logging intended?

Reply:
Yes. The method is only called during initialization, so there won't be too many this log.